ibc-clients/cw-context/src/context/mod.rs

```diff
@@ -191,38 +204,36 @@ impl<'a, C: ClientType<'a>> Context<'a, C> {
     pub fn get_metadata(&self) -> Result<Option<Vec<GenesisMetadata>>, ContractError> {
         let mut metadata = Vec::<GenesisMetadata>::new();
 
-        let start_key = ITERATE_CONSENSUS_STATE_PREFIX.to_string().into_bytes();
-
-        let iterator = self
-            .storage_ref()
-            .range(Some(&start_key), None, Order::Ascending);
-
-        for (_, encoded_height) in iterator {
-            let height = parse_height(encoded_height)?;
-
-            match height {
-                Some(height) => {
-                    let processed_height_key = self.client_update_height_key(&height);
-                    metadata.push(GenesisMetadata {
-                        key: processed_height_key.clone(),
-                        value: self.retrieve(&processed_height_key)?,
-                    });
-                    let processed_time_key = self.client_update_time_key(&height);
-                    metadata.push(GenesisMetadata {
-                        key: processed_time_key.clone(),
-                        value: self.retrieve(&processed_time_key)?,
-                    });
-                }
-                None => continue,
-            }
+        let iterator = CONSENSUS_STATE_HEIGHT_MAP
+            .keys(self.storage_ref(), None, None, Order::Ascending)
+            .flatten();
+
+        for (rev_number, rev_height) in iterator {
+            let height = Height::new(rev_number, rev_height)?;
+
+            let processed_height_key = self.client_update_height_key(&height);
+            metadata.push(GenesisMetadata {
+                key: processed_height_key.clone(),
+                value: self.retrieve(&processed_height_key)?,
+            });
+            let processed_time_key = self.client_update_time_key(&height);
+            metadata.push(GenesisMetadata {
+                key: processed_time_key.clone(),
+                value: self.retrieve(&processed_time_key)?,
+            });
         }
 
-        let iterator = self
-            .storage_ref()
-            .range(Some(&start_key), None, Order::Ascending);
+        let iterator = CONSENSUS_STATE_HEIGHT_MAP
+            .keys(self.storage_ref(), None, None, Order::Ascending)
+            .flatten();
+
+        for (rev_number, rev_height) in iterator {
+            let height = Height::new(rev_number, rev_height)?;
 
-        for (key, height) in iterator {
-            metadata.push(GenesisMetadata { key, value: height });
+            metadata.push(GenesisMetadata {
+                key: iteration_key(rev_number, rev_height),
+                value: height.encode_vec(),
+            });
```

Comment:
What is the purpose of passing `Height` here?

Reply:
As for the why, haven’t had a chance yet to explore where/how this `GenesisMetaData` gonna be used. But the implementation mirrors [ibc-go](https://github.com/cosmos/ibc-go/blob/5ca37ef6e56a98683cf2b3b1570619dc9b322977/modules/light-clients/07-tendermint/store.go#L104-L112).